### PR TITLE
5190 fix keyboard disappearing

### LINKED ISSUE #5190: Stock take line: editing batch > keyboard disappear
## Describe the bug

When have a long stock take line > Editing the batch (Probably all i would say), The keyboard get disappear automatically.
It probably may also depends on the tablet screen size, Some would fit 4 some may have 8 after adding next batch and editing the Keyboard disappear.

### To reproduce

Steps to reproduce the behaviour:

1. Create a new stock take
2. Add many lines
3. Try editing the batch on the last one

### Expected behaviour

It should not.

### Proposed Solution

TDB

### Version and device info

- App version: 8.6.3
- Tablet model: Emulator
- OS version: Android

### Additional context

Raised from client. read here: https://support.msupply.org.nz/cerb6/profiles/ticket/KS-2514-KM
Video: see here: https://github.com/openmsupply/open-msupply-internal/issues/28

### PR DESCRIPTION
**Pull request by raviSussol**
Fixes #5190 

## Change summary
- Introduced `<DataTableView>` component to handle smooth scrollable effect with async rendering its children (virtualized list)
- Wrapped `<DataTable />` with `<DataTableView>` to handle keyboard disappearing in all affected areas
- Refactor keyboard hook
- Updated keyboard visible handler with new `useKeyboard` hook

### Setup
Steps to reproduce or otherwise test the changes of this PR:

- Create a stocktake with many lines so that you have to scroll to view/edit last batch line record
- Try editing any line with scrolling back and forth

### Tests
- [ ] No keyboard disappearing

### Related areas to think about
Make sure to test all other datatable form as well

### CODE REVIEW

File: src/widgets/modals/StocktakeBatchModal.js [hunk 1]
```diff
@@ -269,6 +270,17 @@ export const StocktakeBatchModalComponent = ({
     pageTopLeftSectionContainer,
     pageTopRightSectionContainer,
   } = globalStyles;
+
+  useEffect(() => {
+    // Suppress console error only for this component in production
+    // eslint-disable-next-line no-undef
+    console.reportErrorsAsExceptions = __DEV__;
```
Comment (mark-prins) [inline src/widgets/modals/StocktakeBatchModal.js:277]: what are the console errors you're getting?

**Review by mark-prins — COMMENTED** [review]
Looks fine to me! nice idea adding in the ScrollView
I'm having trouble building, so can't test right now. will have to try again tomorrow

File: src/widgets/modals/StocktakeBatchModal.js (hunk 1 above)
Comment (raviSussol) [reply src/widgets/modals/StocktakeBatchModal.js:277]: Screenshot 2023-05-24 at 1 11 43 PM

**Review by bijaySussol — APPROVED** [review]
@raviSussol :

I am happy with the changes as this works 100% on the area where the problem is.
what also gave me confidence is we have used the ScrollView in the past
I have tested this on Emulator, works very well.

So, I approved... But before merging it, We will need to wait for @mark-prins thoughts.
For me, Works
Approved and good work @raviSussol .

**Review by bijaySussol — DISMISSED** [review]
@raviSussol :

I am happy with the changes as this works 100% on the area where the problem is.
what also gave me confidence is we have used the ScrollView in the past
I have tested this on Emulator, works very well.

So, I approved... But before merging it, We will need to wait for @mark-prins thoughts.
For me, Works
Approved and good work @raviSussol .

**Comment by bijaySussol** [comment]
Got this notification:

photo_2023-05-24_13-27-51

**Review by bijaySussol — CHANGES_REQUESTED** [review]
@raviSussol : 

Tested on Emulator.
I see the list is not been refreshed
See here: https://github.com/openmsupply/open-msupply-internal/issues/28#issuecomment-1562840010 for the images

With that requesting changes

**Comment by raviSussol** [comment]
> @raviSussol :
> 
> Tested on Emulator. I see the list is not been refreshed See here: [openmsupply/open-msupply-internal#28 (comment)](https://github.com/openmsupply/open-msupply-internal/issues/28#issuecomment-1562840010) for the images
> 
> With that requesting changes

4bf924d8b should fix that.

**Comment by bijaySussol** [comment]
> > @raviSussol :
> > Tested on Emulator. I see the list is not been refreshed See here: [openmsupply/open-msupply-internal#28 (comment)](https://github.com/openmsupply/open-msupply-internal/issues/28#issuecomment-1562840010) for the images
> > With that requesting changes
> 
> [4bf924d](https://github.com/openmsupply/mobile/commit/4bf924d8be8bc6781f9bfae9df690a2c9bbc9163) should fix that.

It does indeed thanks

**Review by bijaySussol — DISMISSED** [review]
@raviSussol : 

From my testing point of view, It is working 100%
so, Green light from my side and i am happy to approved
Approved!

**Review by arjunSussol — APPROVED** [review]
@raviSussol looks fine, both tested and reviewed!

**Pull request merged**